book_source/03_topical_pages/07_remote_access/01_pecan_api.Rmd

```diff
@@ -43,14 +43,26 @@ The currently implemented functionalities include:
   * [`GET /api/pfts/`](#get-apipfts): Search for PFT(s) using search pattern based on PFT name, PFT type & Model type
   * [`GET /api/pfts/{pft_id}`](#get-apipftspft_id): Fetch the details of specific PFT
 
+* __Formats:__
+  * [`GET /api/formats/`](#get-apiformats): Search for format(s) using search pattern based on format name & mime type
+  * [`GET /api/formats/{format_id}`](#get-formatsformat_id): Fetch the details of specific Format
+  
+* __Inputs:__
+  * [`GET /api/inputs/`](#get-apiinputs): Search for inputs needed for a PEcAn workflow based on `model_id` & `site_id`
+  * [`GET /api/inputs/{input_id}`](#get-apiinputsinput_id) *: Download the desired input file
```

Comment:
Umm...Currently I implemented it in a way assuming that the input_id points to a file & not a folder. Also, the current VM that I am working on doesn't seem to have folders containing multiple met files. So would be difficult to test. Can you suggest some way?

Reply:
@robkooper I'm not sure what's on the machine @tezansahu is testing the API on, but could you point him to an example CF met Input record where the files are annual.

@tezansahu if the API system only allows one file to be returned at a time, then I think you'd want to have an optional way to get file by name or by order in a sequence. For the latter I'm thinking something similar to how the search functions will return the first N queries, then you query again with an offset to get the next N queries. But that options doesn't seem great to me because you wouldn't really know which file you're getting so you wouldn't know how to name them. So I guess the simplest place to start (e.g. to get this PR in) is to just give a check on the number of files in an Input and if it's >1 and the file isn't requested by name then we need to throw an error.

Reply:
I have addressed this in the latest commits